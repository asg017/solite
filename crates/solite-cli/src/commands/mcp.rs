use crate::cli::{McpCommand, McpNamespace};


pub(crate) fn mcp(cmd: McpNamespace) -> Result<(), ()> {
  match cmd.command {
    McpCommand::Up(_) => solite_mcp::upp().map_err(|_| ()),
    McpCommand::Install(_) => {
      todo!("MCP install is not implemented yet.")
    }
  }
}