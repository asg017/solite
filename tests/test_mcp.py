import pytest
from mcp import McpError
from fastmcp import Client

class TestMcp:
    def __init__(self, mcp_client: Client, snapshot):
        self.mcp_client = mcp_client
    
    async def execute_sql(self, sql: str, snapshot=None, name: str | None = None):
        result = await self.mcp_client.call_tool("execute_sql", {"sql": sql})
        if snapshot is not None:
            text_content = "\n".join([x.text for x in result.content])
            assert text_content == snapshot(name=name)
        return result

@pytest.mark.asyncio
async def test_mcp(mcp_client: Client, snapshot):
    tools = await mcp_client.list_tools()
    assert sorted([tool.name for tool in tools]) == snapshot(name="list_tools")
    resources = await mcp_client.list_resources()
    assert resources == snapshot(name="list_resources")
    prompts = await mcp_client.list_prompts()
    assert prompts == snapshot(name="list_prompts")

    t = TestMcp(mcp_client, snapshot)
    await t.execute_sql("select 1 + 1", snapshot=snapshot, name="select 1 + 1")
    await t.execute_sql("create table t as select value from json_each('[1, 2, 3]');")
    await t.execute_sql("select * from t", snapshot=snapshot, name="select * from t")
    
    with pytest.raises(McpError):
        await t.execute_sql("select error;")