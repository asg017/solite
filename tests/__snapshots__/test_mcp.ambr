# serializer version: 1
# name: test_mcp[list_prompts]
  list([
  ])
# ---
# name: test_mcp[list_resources]
  list([
  ])
# ---
# name: test_mcp[list_tools]
  list([
    'echo',
    'execute_sql',
    'export_database',
  ])
# ---
# name: test_mcp[select * from t]
  '''
  value: 1
  value: 2
  value: 3
  
  '''
# ---
# name: test_mcp[select 1 + 1]
  '''
  1 + 1: 2
  
  '''
# ---
